catsim/estimation.py

```diff
@@ -240,3 +241,136 @@ def estimate(
         self._evaluations = res.nfev
 
         return res.x[0]
+
+class BayesianEstimator(Estimator):
+    """Estimator that uses Bayesian estimation for ability estimation
+
+    :param prior_mean: mean of the prior distribution
+    :param prior_sd: standard deviation of the prior distribution
+    :param bounds: bounds of the ability estimate
+    :param verbose: verbosity level of the estimation
+    """
+
+    def __str__(self):
+        return 'Bayesian Estimator'
+
+    def __init__(self,
+            precision: int = 6,
+            prior_mean: float = 0.,
+            prior_sd: float = 1.,
+            bounds: tuple = (-6, 6),
+            verbose: bool = False
+            ):
+        super().__init__()
+        self._precision = precision
+        self._prior_mean = prior_mean
+        self._prior_sd = prior_sd
+        self._verbose = verbose
+        self._evaluations = 0
+        self._calls = 0
+        self._bounds = bounds
+
+    @property
+    def calls(self) -> float:
+        """How many times the estimator has been called to maximize/minimize the log-likelihood function
+
+        :returns: number of times the estimator has been called to maximize/minimize the log-likelihood function"""
+        return self._calls
+
+    @property
+    def evaluations(self) -> float:
+        """Total number of times the estimator has evaluated the log-likelihood function during its existence
+
+        :returns: number of function evaluations"""
+        return self._evaluations
+
+    @property
+    def avg_evaluations(self) -> float:
+        """Average number of function evaluations for all tests the estimator has been used
+
+        :returns: average number of function evaluations"""
+        return self._evaluations / self._calls
+
+    def _bound_estimate(self, estimate):
+        lbound, ubound = self._bounds
+        return min(max(estimate, lbound), ubound)
+
+    def estimate(
+        self,
+        index: int = None,
+        items: numpy.ndarray = None,
+        administered_items: list = None,
+        response_vector: list = None,
+        est_theta: float = None,
```

Comment:
Why is the current est_theta passed in? It is unused in this case since we are recalculating the entire posterior and not doing bayesian updating

Reply:
Good catch, it used to be in for the dodd correction, but we no longer need that here.